File: Scripts/HealingBed.cs
using UnityEngine;
using System.Collections;

public class HealingBed : MonoBehaviour {
	public float amount = 170;  //default to 2/3 of 255, the total player can have
	public int minSecurityLevel = 0;
	public bool broken = false;
	private float give;

	private AudioSource SFXSource;

	void Awake () {
		SFXSource = GetComponent<AudioSource>();
	}

	public void Use (UseData ud) {
		if (GetCurrentLevelSecurity() <= minSecurityLevel) {
			if (!broken) {
				PlayerReferenceManager.a.playerCapsule.GetComponent<HealthManager>().HealingBed(amount,true);
				PlayerHealth.a.radiationArea = false;
				PlayerHealth.a.radiated = 0f;
				CenterStatusPrint("%s", Eng_Text->stringTable[23],ud.owner);
				Utils.PlayOneShotSavable(SFXSource,sounds[103]);
			} else {
				CenterStatusPrint("%s", Eng_Text->stringTable[24],ud.owner);
			}
		} else {
			Eng_UI->BlockedBySecurity(Eng_Global->instances[i].position);
		}
	}
}

File: Scripts/healingbed.c
#include "mod.h"

void HealingBedUse(uint16_t self, uint16_t owner) {
    Entity* e = &Eng_Global->instances[self];
    if (GetCurrentLevelSecurity() <= (uint8_t)e->minSecurityLevel) {
        if (!e->broken) {
            HealthManagerHealingBed(PLAYER1,e->amount,true);
            CenterStatusPrint("%s",Eng_Text->stringTable[23],owner);
            play_wav(sounds[103],1.0f,e->position,false);
        } else CenterStatusPrint("%s",Eng_Text->stringTable[24],owner);
    } else UIBlockedBySecurity(e->position);
}
